cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramRunStatusMonitorService.java

```diff
@@ -150,11 +159,20 @@ Set<ProgramRunId> terminatePrograms() {
         // If this run was initiated by this CDAP instance to run on another instance, it will use the tethering
         // provisioner. The instance that actually runs it on the other side will have a non-null peer name.
         String provisionerName = SystemArguments.getProfileProvisioner(record.getSystemArgs());
+        ProgramRunId programRunId = record.getProgramRunId();
         if (record.getPeerName() == null && TetheringProvisioner.TETHERING_NAME.equals(provisionerName)) {
+          if (record.getStoppingTs() != null &&
```

Comment:
can this be null if the state is STOPPING?

Reply:
It should not be null, and neither should terminateTs. We were already had a null check for terminateTs (probably just to safe, avoid NullpointerException if the timestamp was somehow null due to a bug), so added a similar null check for stoppingTs as well.